doc/release/1.13.0-notes.rst

```diff
@@ -162,8 +162,26 @@ for ``len(d.shape)``.
 ``np.block`` function for creating blocked arrays
 -------------------------------------------------
 Add a new ``block`` function to the current stacking functions ``vstack``,
-``hstack``, and ``stack``. It is similar to Matlab's square bracket
-notation for creating block matrices.
+``hstack``, and ``stack``. This allows concatenation across multiple axes
+simultaneously, with a similar syntax to array creation, but where elements
+can themselves be arrays. For instance::
+
+    >>> A = np.eye(2) * 2
+    >>> B = np.eye(3) * 3
+    >>> np.block([
+    ...     [A,               np.zeros((2, 3))],
+    ...     [np.ones((3, 2)), B               ]
+    ... ])
+    array([[ 2.,  0.,  0.,  0.,  0.],
+           [ 0.,  2.,  0.,  0.,  0.],
+           [ 1.,  1.,  3.,  0.,  0.],
+           [ 1.,  1.,  0.,  3.,  0.],
+           [ 1.,  1.,  0.,  0.,  3.]])
```

Comment:
Love this example. Particularly after our [discussion]( https://github.com/numpy/numpy/pull/7768#issuecomment-291542328 ). Is there a test for this as well? Just familiarizing myself with this PR now. So if I overlooked it, sorry about that.